problems/problems.go

```diff
@@ -0,0 +1,74 @@
+package errors
+
+import (
+	"fmt"
+)
+
+//Problem implements the RFC7807 "problem"/error standard
+//Additional error field can be defined in custom struct implementing the Problem struct
+type Problem struct {
+	Title    string `json:"title"`
+	Detail   string `json:"detail,omitempty"`
+	Type     string `json:"type,omitempty"`
+	Instance string `json:"instance,omitempty"`
+	Status   int    `json:"status,omitempty"`
+}
+
+//Error implements the error interface
+func (p *Problem) Error() string {
+	return fmt.Sprintf("Title: %s; Detail: %s", p.Title, p.Detail)
+}
+
+//DefaultType implements the default type content described in RFC7807
+const DefaultType = "about:blank"
+
+//SetTitle sets the title of the problem object
+func (p *Problem) SetTitle(title string) *Problem {
+	p.Title = title
+	return p
+}
+
+//SetDetail sets the detail info of the problem object
```

Comment:
Document meaning not code.
If somebody doesn't know the RFC Spec it would be great if the comment is sufficient to know what all this is for.

In general comments should less explain what the code does but what it means (and why).

Just a nit/tip :-)